pkg/backend/stack.go

```diff
@@ -171,7 +172,8 @@ func GetEnvironmentTagsForCurrentStack() (map[apitype.StackTagName]string, error
 	// Tags based on Pulumi.yaml.
 	projPath, err := workspace.DetectProjectPath()
 	if err != nil {
-		return nil, err
+		// No current stack return empty
+		return make(map[apitype.StackTagName]string), nil
```

Comment:
has more logical error handling imo than having the caller determine empty tagset.